packages/token/src/tests/erc2981/test_erc2981_ownable.cairo

```diff
@@ -0,0 +1,270 @@
+use crate::common::erc2981::ERC2981Component;
+use crate::common::erc2981::interface::{IERC2981ABIDispatcher, IERC2981ABIDispatcherTrait};
+use openzeppelin_testing as utils;
+use openzeppelin_testing::constants::{OWNER, OTHER, ZERO, RECIPIENT};
+use openzeppelin_utils::serde::SerializedAppend;
+use snforge_std::start_cheat_caller_address;
+use starknet::{ContractAddress, contract_address_const};
+
+fn DEFAULT_RECEIVER() -> ContractAddress {
+    contract_address_const::<'DEFAULT_RECEIVER'>()
+}
+
+// 0.5% (default denominator is 10000)
+const DEFAULT_FEE_NUMERATOR: u128 = 50;
+// 5% (default denominator is 10000)
+const FEE_NUMERATOR: u128 = 500;
+
+fn setup_dispatcher() -> IERC2981ABIDispatcher {
```

Comment:
Similar to the previous comment, the COMPONENT_STATE pattern should be used for this.

Reply:
I think in this specific case it's better to use dispatcher-based approach in order to make it easier to differentiate between external and internal functions with same names

Reply:
https://github.com/OpenZeppelin/cairo-contracts/pull/1173#discussion_r1806259337

Reply:
Refactored to COMPONENT_STATE